code/modules/hydroponics/seed_machines.dm

```diff
@@ -42,14 +42,27 @@
 	var/eject_disk = 0
 	var/failed_task = 0
 	var/disk_needs_genes = 0
+	var/time_coeff = 1
+	var/degradation_coeff = 1
+
+/obj/machinery/botany/RefreshParts()
+	var/T = 0
+	for(var/obj/item/weapon/stock_parts/micro_laser/ML in component_parts)
+		T += ML.rating
+	degradation_coeff = round(T/2)
+	T = 0
+	for(var/obj/item/weapon/stock_parts/manipulator/MA in component_parts)
+		T += MA.rating
+	time_coeff = T
+	T = 0
```

Comment:
This isn't entirely necessary. Were you planning on adding something for the scanning modules to do?

Reply:
Just copying how the autolathe's RefreshParts() was written; didn't think it was necessary here but it coulda been an idiosyncrasy of the language/function.